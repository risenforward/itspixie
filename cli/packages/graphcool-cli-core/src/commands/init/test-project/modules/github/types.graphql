type GithubUser {
  id: ID! @isUnique
  createdAt: DateTime!
  githubUserId: String @isUnique
  updatedAt: DateTime!
}
