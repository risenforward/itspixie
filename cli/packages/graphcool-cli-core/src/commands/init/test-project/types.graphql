# This file contains the GraphQL Types
# Graphcool has one special type, the File type:

# type File implements Node {
#   contentType: String!
#   createdAt: DateTime!
#   id: ID! @isUnique
#   name: String!
#   secret: String! @isUnique
#   size: Int!
#   updatedAt: DateTime!
#   url: String! @isUnique
# }

# All types need to have an id field defined like this:

# type User implements Node {
#   id: ID! @isUnique
# }

