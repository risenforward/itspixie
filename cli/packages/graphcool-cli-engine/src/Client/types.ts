export interface User {
  id: string
  name: string
  login: Login[]
}

export interface Login {
  email: string
}
