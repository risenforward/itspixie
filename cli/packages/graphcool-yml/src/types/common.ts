export interface Args {
  [name: string]: string | boolean
}

export interface Stages {
  [name: string]: string
}
