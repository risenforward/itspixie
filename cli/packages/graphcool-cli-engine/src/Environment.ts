import { Output } from './index'
import { Config } from './Config'
import { Args } from './types/common'
import { Cluster } from './Cluster'
import * as fs from 'fs-extra'
import * as yaml from 'js-yaml'
import Variables from './GraphcoolDefinition/Variables'
import { InternalRC } from './types/rc'
import { ClusterNotFound } from './errors/ClusterNotFound'

export class Environment {
  out: Output
  config: Config
  args: Args
  activeCluster: Cluster
  globalRC: InternalRC = {}
  clusters: Cluster[]
  platformToken?: string
  constructor(out: Output, config: Config) {
    this.out = out
    this.config = config
  }

  async load(args: Args) {
    await this.loadGlobalRC()
  }

  clusterByName(name: string, throws: boolean = false): Cluster | undefined {
    const cluster = this.clusters.find(c => c.name === name)
    if (!throws) {
      return cluster
    }

    if (!cluster) {
      throw new ClusterNotFound(name)
    }

    return cluster
  }

  setToken(token: string | undefined) {
    this.globalRC.platformToken = token
  }

  addCluster(cluster: Cluster) {
    this.clusters.push(cluster)
  }

  saveGlobalRC() {
    const rc = {
      platformToken: this.globalRC.platformToken,
      clusters: this.getLocalClusterConfig(),
    }
    const rcString = yaml.safeDump(rc)
    fs.writeFileSync(this.config.globalRCPath, rcString)
  }

  setActiveCluster(cluster: Cluster) {
    this.activeCluster = cluster
  }

  private async loadGlobalRC(): Promise<void> {
    const globalFile =
      this.config.globalRCPath && fs.pathExistsSync(this.config.globalRCPath)
        ? fs.readFileSync(this.config.globalRCPath, 'utf-8')
        : null
    if (globalFile) {
      this.globalRC = await this.loadYaml(globalFile, this.config.globalRCPath)
      this.clusters = this.initClusters(this.globalRC)
      this.platformToken =
        this.globalRC.platformToken || process.env.GRAPHCOOL_PLATFORM_TOKEN
    }
  }

  private async loadYaml(
    file: string | null,
    filePath: string | null = null,
  ): Promise<any> {
    if (file) {
      let content
      try {
        content = yaml.safeLoad(file)
      } catch (e) {
        this.out.error(`Yaml parsing error in ${filePath}: ${e.message}`)
      }
      const variables = new Variables(
        this.out,
        filePath || 'no filepath provided',
        this.args,
      )
      content = await variables.populateJson(content)

      return content
    } else {
      return {}
    }
  }

  private getClustersFromRC(rc: InternalRC): Cluster[] {
    if (!rc.clusters) {
      return []
    }
    return Object.keys(rc.clusters).map(name => {
      const cluster = rc.clusters![name]
      return new Cluster(name, cluster.host, cluster.clusterSecret, true)
    })
  }

  private initClusters(rc: InternalRC): Cluster[] {
    const rcClusters = this.getClustersFromRC(rc)
    const sharedClusters = this.getSharedClusters(rc)
    return [...rcClusters, ...sharedClusters]
  }

  private getSharedClusters(rc: InternalRC): Cluster[] {
    return this.config.sharedClusters.map(clusterName => {
      return new Cluster(
        clusterName,
        this.config.sharedEndpoint,
        rc.platformToken!,
        false,
      )
    })
  }

  private getLocalClusterConfig() {
    return this.clusters.filter(c => c.local).reduce((acc, cluster) => {
      return {
        ...acc,
        [cluster.name]: {
          host: cluster.baseUrl,
          clusterSecret: cluster.token,
        },
      }
    }, {})
  }
}
