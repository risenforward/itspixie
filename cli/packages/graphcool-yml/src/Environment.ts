import { Args } from './types/common'
import { Cluster } from './Cluster'
import * as fs from 'fs-extra'
import * as yaml from 'js-yaml'
import { ClusterNotFound } from './errors/ClusterNotFound'
import { Variables } from './Variables'
import { IOutput, Output } from './Output'
import * as path from 'path'
import * as os from 'os'
import chalk from 'chalk'
import 'isomorphic-fetch'
import { RC } from './index'
import { DatabaseRC } from './types/rc'
import { EnvironmentMigrator } from './EnvironmentMigrator'
const debug = require('debug')('Environment')

const isDev = (process.env.ENV || '').toLowerCase() === 'dev'

export class Environment {
  sharedClusters: string[] = [
    'graphcool-eu1',
    'graphcool-us1',
    'shared-public-demo',
  ]
  clusterEndpointMap: { [key: string]: string } = {
    'graphcool-eu1': 'https://graphcool-eu1.graphcool.cloud',
    'graphcool-us1': 'https://graphcool-us1.graphcool.cloud',
    'shared-public-demo': isDev
      ? 'https://dev.database-beta.graph.cool'
      : 'https://database-beta.graph.cool',
  }
  args: Args
  activeCluster: Cluster
  globalRC: RC = {}
  databaseRC: DatabaseRC = {}
  clusters: Cluster[]
  out: IOutput
  home: string
  rcPath: string
  constructor(home: string, out: IOutput = new Output()) {
    this.out = out
    this.home = home

    this.rcPath = path.join(this.home, '.graphcoolrc')
    const migrator = new EnvironmentMigrator(home, out)
    debug(`migrating`)
    migrator.migrate()
  }

  async load(args: Args) {
    await Promise.all([this.loadGlobalRC(), this.setSharedClusters()])
  }

  async setSharedClusters() {
    // TODO: reenable

    try {
      const res = await fetch('https://stats.graph.cool/', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        } as any,
        body: JSON.stringify({
          query: `
        {
          publicClusters {
            name
            endpoint
            description
          }
        }
        `,
        }),
      })
      const json = await res.json()
      if (
        json &&
        json.data &&
        json.data.publicClusters &&
        Array.isArray(json.data.publicClusters) &&
        json.data.publicClusters.length > 0
      ) {
        this.sharedClusters = json.data.publicClusters.map(c => c.name)
        this.clusterEndpointMap = json.data.publicClusters.reduce(
          (acc, curr) => {
            return {
              ...acc,
              [curr.name]: curr.endpoint,
            }
          },
          {},
        )

        debug(this.sharedClusters)
        debug(this.clusterEndpointMap)
      }
    } catch (e) {
      debug(e)
      //
    }
  }

  clusterByName(name: string, throws: boolean = false): Cluster | undefined {
    const cluster = this.clusters.find(c => c.name === name)
    if (!throws) {
      return cluster
    }

    if (!cluster) {
      throw new ClusterNotFound(name)
    }

    return cluster
  }

  setToken(token: string | undefined) {
    this.databaseRC.cloudSessionKey = token
  }

  addCluster(cluster: Cluster) {
    const existingClusterIndex = this.clusters.findIndex(
      c => c.name === cluster.name,
    )
    if (existingClusterIndex > -1) {
      this.clusters.splice(existingClusterIndex, 1)
    }
    this.clusters.push(cluster)
  }

  saveGlobalRC() {
    const rc = {
      'graphcool-1.0': {
        cloudSessionKey: this.databaseRC.cloudSessionKey,
        clusters: this.getLocalClusterConfig(),
      },
    }
    if (this.globalRC['graphcool-framework']) {
      rc['graphcool-framework'] = this.globalRC['graphcool-framework']
    }
    // parse & stringify to rm undefined for yaml parser
    const rcString = yaml.safeDump(JSON.parse(JSON.stringify(rc)))
    fs.writeFileSync(this.rcPath, rcString)
  }

  setActiveCluster(cluster: Cluster) {
    this.activeCluster = cluster
  }

  async loadGlobalRC(): Promise<void> {
    const globalFile =
      this.rcPath && fs.pathExistsSync(this.rcPath)
        ? fs.readFileSync(this.rcPath, 'utf-8')
        : undefined
    await this.parseGlobalRC(globalFile)
  }

  async parseGlobalRC(globalFile?: string): Promise<void> {
    if (globalFile) {
      this.globalRC = await this.loadYaml(globalFile, this.rcPath)
      this.databaseRC = this.globalRC['graphcool-1.0'] || {}
    }
    this.clusters = this.initClusters(this.databaseRC)
  }

  private async loadYaml(
    file: string | null,
    filePath: string | null = null,
  ): Promise<any> {
    if (file) {
      let content
      try {
        content = yaml.safeLoad(file)
      } catch (e) {
        throw new Error(`Yaml parsing error in ${filePath}: ${e.message}`)
      }
      const variables = new Variables(
        filePath || 'no filepath provided',
        this.args,
        this.out,
      )
      content = await variables.populateJson(content)

      return content
    } else {
      return {}
    }
  }

  private initClusters(rc: DatabaseRC): Cluster[] {
    const rcClusters = this.getClustersFromRC(rc)
    const sharedClusters = this.getSharedClusters(rc)
    return [...rcClusters, ...sharedClusters]
  }

  private getSharedClusters(rc: DatabaseRC): Cluster[] {
    return this.sharedClusters.map(clusterName => {
      return new Cluster(
        clusterName,
        this.clusterEndpointMap[clusterName],
        rc && rc.cloudSessionKey,
        false,
      )
    })
  }

  private getClustersFromRC(rc: DatabaseRC): Cluster[] {
    if (!rc.clusters) {
      return []
    }
    return Object.keys(rc.clusters).map(name => {
      const cluster = rc.clusters![name]
      return new Cluster(name, cluster.host, cluster.clusterSecret, true)
    })
  }

  private getLocalClusterConfig() {
    return this.clusters.filter(c => c.local).reduce((acc, cluster) => {
      return {
        ...acc,
        [cluster.name]: {
          host: cluster.baseUrl,
          clusterSecret: cluster.clusterSecret,
        },
      }
    }, {})
  }
}
