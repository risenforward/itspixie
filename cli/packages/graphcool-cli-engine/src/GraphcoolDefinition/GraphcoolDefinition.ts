import { Config } from '../Config'
import { Output } from '../index'
import { readDefinition } from './yaml'
import { Args, Stages } from '../types/common'
import { GraphcoolDefinition } from 'graphcool-json-schema'
import * as fs from 'fs-extra'
import chalk from 'chalk'
import { Environment } from '../Environment'
import { mapValues } from 'lodash'
import * as yamlParser from 'yaml-ast-parser'
import { StageNotFound } from '../errors/StageNotFound'

interface ErrorMessage {
  message: string
}

export class GraphcoolDefinitionClass {
  out: Output
  config: Config
  definition?: GraphcoolDefinition
  typesString?: string
  rawStages: Stages
  private definitionString: string
  constructor(out: Output, config: Config) {
    this.out = out
    this.config = config
  }
  async load(env: Environment, args: Args) {
    if (this.config.definitionPath) {
      this.definition = await readDefinition(
        this.config.definitionPath,
        this.out,
        args,
      )
      this.definitionString = fs.readFileSync(
        this.config.definitionPath,
        'utf-8',
      )
      this.rawStages = this.definition.stages
      this.definition.stages = this.resolveStageAliases(this.definition.stages)
      this.ensureOfClusters(this.definition, env)
      this.typesString = this.getTypesString(this.definition)
    } else {
      throw new Error(`Please create a graphcool.yml`)
    }
  }

  getStage(name: string, throws: boolean = false): string | undefined {
    const stage =
      this.definition &&
      (this.definition.stages[name] || this.definition.stages.default)
    if (!throws) {
      return stage
    }

    if (!stage) {
      throw new StageNotFound(name)
    }

    return stage
  }

  setStage(name: string, clusterName: string) {
    this.definitionString = this.insertToDefinition(
      this.definitionString,
      'stages',
      `\n  ${name}: ${clusterName}`,
    )
  }

  insertToDefinition(file: string, key: string, insertion: string) {
    const obj = yamlParser.safeLoad(file)

    const mapping = obj.mappings.find(m => m.key.value === key)
    if (mapping) {
      const end = mapping.endPosition

      const newFile = file.slice(0, end) + insertion + file.slice(end)
      const valueStart = mapping.value.startPosition
      const valueEnd = mapping.value.endPosition
      if (mapping.value && valueEnd - valueStart < 4) {
        return newFile.slice(0, valueStart) + newFile.slice(valueEnd)
      }

      return newFile
    } else {
      return file + `\n${key}: ` + insertion
    }
  }

  save() {
    fs.writeFileSync(this.config.definitionPath!, this.definitionString)
  }

  private getTypesString(definition: GraphcoolDefinition) {
    const typesPaths = Array.isArray(definition.datamodel)
      ? definition.datamodel
      : [definition.datamodel]

    const errors: ErrorMessage[] = []
    let allTypes = ''
    typesPaths.forEach(typesPath => {
      if (fs.existsSync(typesPath)) {
        const types = fs.readFileSync(typesPath, 'utf-8')
        allTypes += types + '\n'
      } else {
        throw new Error(
          `The types definition file "${typesPath}" could not be found.`,
        )
      }
    })

    return allTypes
  }

  private ensureOfClusters(definition: GraphcoolDefinition, env: Environment) {
    Object.keys(definition.stages).forEach(stageName => {
      const referredCluster = definition.stages[stageName]
      if (!env.clusters.find(c => c.name === referredCluster)) {
        throw new Error(
          `Could not find cluster '${
            referredCluster
          }', which is used in stage '${stageName}'.`,
        )
      }
    })
  }

  private resolveStageAliases = stages =>
    mapValues(stages, target => this.resolveStage(target, stages))

  private resolveStage = (
    stage: string,
    stages: { [key: string]: string },
  ): string =>
    stages[stage] ? this.resolveStage(stages[stage], stages) : stage

  get default(): string | null {
    if (
      this.definition &&
      this.definition.stages &&
      this.definition.stages.default
    ) {
      return this.definition.stages.default
    }

    return null
  }
}
