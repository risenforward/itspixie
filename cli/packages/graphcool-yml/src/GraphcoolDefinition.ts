import { readDefinition } from './yaml'
import { GraphcoolDefinition } from 'graphcool-json-schema'
import * as fs from 'fs-extra'
import { mapValues } from 'lodash'
import * as yamlParser from 'yaml-ast-parser'
import * as dotenv from 'dotenv'
import * as path from 'path'
import * as jwt from 'jsonwebtoken'
import { Args } from './types/common'
import { StageNotFound } from './errors/StageNotFound'
import { Environment } from './Environment'
import { IOutput } from './Output'
import { Cluster } from './Cluster'

interface ErrorMessage {
  message: string
}

export interface EnvVars {
  [key: string]: string | undefined
}

export class GraphcoolDefinitionClass {
  definition?: GraphcoolDefinition
  typesString?: string
  secrets: string[] | null
  definitionPath?: string | null
  definitionDir: string
  env: Environment
  out?: IOutput
  envVars: any
  private definitionString: string
  constructor(
    env: Environment,
    definitionPath?: string | null,
    envVars: EnvVars = process.env,
    out?: IOutput,
  ) {
    this.secrets = null
    this.definitionPath = definitionPath
    if (definitionPath) {
      this.definitionDir = path.dirname(definitionPath)
    }
    this.env = env
    this.out = out
    this.envVars = envVars
  }
  async load(args: Args, envPath?: string) {
    dotenv.config({ path: envPath })
    if (this.definitionPath) {
      this.definition = await readDefinition(
        this.definitionPath,
        args,
        this.out,
      )
      this.definitionString = fs.readFileSync(this.definitionPath, 'utf-8')
      this.typesString = this.getTypesString(this.definition)
      const secrets = this.definition.secret
      this.secrets = secrets ? secrets.replace(/\s/g, '').split(',') : null
      const disableAuth = this.definition.disableAuth
      if (this.secrets === null && !disableAuth) {
        throw new Error(
          'Please either provide a secret in your graphcool.yml or disableAuth: true',
        )
      }
    }
  }

  getToken(serviceName: string, stageName: string): string | undefined {
    if (this.secrets) {
      const data = {
        data: {
          service: `${serviceName}@${stageName}`,
          roles: ['admin'],
        },
      }
      return jwt.sign(data, this.secrets[0], {
        expiresIn: '7d',
      })
    }

    return undefined
  }

  getCluster(): Cluster | undefined {
    if (this.definition && this.definition.cluster) {
      return this.env.clusterByName(this.definition.cluster)
    }

    return undefined
  }

  getTypesString(definition: GraphcoolDefinition) {
    const typesPaths = Array.isArray(definition.datamodel)
      ? definition.datamodel
      : [definition.datamodel]

    const errors: ErrorMessage[] = []
    let allTypes = ''
    typesPaths.forEach(unresolvedTypesPath => {
      const typesPath = path.join(this.definitionDir, unresolvedTypesPath)
      if (fs.existsSync(typesPath)) {
        const types = fs.readFileSync(typesPath, 'utf-8')
        allTypes += types + '\n'
      } else {
        throw new Error(
          `The types definition file "${typesPath}" could not be found.`,
        )
      }
    })

    return allTypes
  }
}
