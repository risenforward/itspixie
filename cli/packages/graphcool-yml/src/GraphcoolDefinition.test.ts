import { GraphcoolDefinitionClass } from './GraphcoolDefinition'
import * as fs from 'fs-extra'
import * as path from 'path'
import { getTmpDir } from './test/getTmpDir'
import { Environment } from './Environment'
import { makeEnv } from './Environment.test'
import { Args } from './types/common'

const defaultGlobalRC = `clusters:
local:
  host: 'http://localhost:60000'
remote:
  host: 'https://remote.graph.cool'
  clusterSecret: 'here-is-a-token'
`

function makeDefinition(
  yml: string,
  datamodel: string,
  args: Args = {},
  globalRC: string = defaultGlobalRC,
) {
  const definitionDir = getTmpDir()
  const definitionPath = path.join(definitionDir, 'graphcool.yml')
  const modelPath = path.join(definitionDir, 'datamodel.graphql')
  const env = makeEnv(defaultGlobalRC)

  const definition = new GraphcoolDefinitionClass(env, definitionPath)

  fs.writeFileSync(modelPath, datamodel)
  fs.writeFileSync(definitionPath, yml)

  return { env, definition }
}

async function loadDefinition(
  yml: string,
  datamodel: string,
  args: Args = {},
  envPath?: string,
  globalRC: string = defaultGlobalRC,
) {
  const { env, definition } = makeDefinition(yml, datamodel, args, globalRC)
  await env.load(args)
  await definition.load(args, envPath)
  return { env, definition }
}

describe('graphcool definition', () => {
  test('load basic yml', async () => {
    const yml = `\
service: jj

datamodel:
- datamodel.graphql

secret: some-secret

schema: schemas/database.graphql
stages: 
  default: dev
  dev: shared-eu-west-1
    `
    const datamodel = `
type User @model {
  id: ID! @isUnique
  name: String!
  lol: Int
  what: String
}
`

    const { definition } = await loadDefinition(yml, datamodel)

    expect(definition.definition).toMatchSnapshot()
  })
  test('load yml with secret and env var', async () => {
    const secret = 'this-is-a-long-secret'
    process.env.MY_TEST_SECRET = secret
    const yml = `\
service: jj

datamodel:
- datamodel.graphql

secret: \${env:MY_TEST_SECRET}

schema: schemas/database.graphql
stages: 
  default: dev
  dev: shared-eu-west-1
    `
    const datamodel = `
type User @model {
  id: ID! @isUnique
  name: String!
  lol: Int
  what: String
}
`

    const { definition } = await loadDefinition(yml, datamodel)

    expect(definition.definition).toMatchSnapshot()
  })
  test('load yml with secret and env var in .env', async () => {
    const secret = 'this-is-a-long-secret'
    const yml = `\
service: jj

datamodel:
- datamodel.graphql

secret: \${env:MY_DOT_ENV_SECRET}

schema: schemas/database.graphql
stages: 
  default: dev
  dev: shared-eu-west-1
    `
    const datamodel = `
type User @model {
  id: ID! @isUnique
  name: String!
  lol: Int
  what: String
}
`
    const { definition, env } = makeDefinition(yml, datamodel, {})
    const envPath = path.join(definition.definitionDir, '.env')

    fs.writeFileSync(
      envPath,
      `MY_DOT_ENV_SECRET=this-is-very-secret,and-comma,seperated`,
    )

    await env.load({})
    await definition.load({}, envPath)

    expect(definition.definition).toMatchSnapshot()
    expect(definition.secrets).toMatchSnapshot()
  })
  test('load yml with disableAuth: true', async () => {
    const secret = 'this-is-a-long-secret'
    const yml = `\
service: jj

datamodel:
- datamodel.graphql

disableAuth: true

schema: schemas/database.graphql
stages: 
  default: dev
  dev: shared-eu-west-1
    `
    const datamodel = `
type User @model {
  id: ID! @isUnique
  name: String!
  lol: Int
  what: String
}
`
    const { definition, env } = makeDefinition(yml, datamodel)
    const envPath = path.join(definition.definitionDir, '.env')

    fs.writeFileSync(
      envPath,
      `MY_DOT_ENV_SECRET=this-is-very-secret,and-comma,seperated`,
    )

    await env.load({})
    await definition.load({}, envPath)

    expect(definition.definition).toMatchSnapshot()
    expect(definition.secrets).toMatchSnapshot()
  })
  test('load basic yml', async () => {
    const yml = `\
service: jj

datamodel:
- datamodel.graphql

schema: schemas/database.graphql
stages: 
  default: dev
  dev: shared-eu-west-1
    `
    const datamodel = `
type User @model {
  id: ID! @isUnique
  name: String!
  lol: Int
  what: String
}
`

    let error
    try {
      const { definition } = await loadDefinition(yml, datamodel)
    } catch (e) {
      error = e
    }

    expect(error).toMatchSnapshot()
  })
})
