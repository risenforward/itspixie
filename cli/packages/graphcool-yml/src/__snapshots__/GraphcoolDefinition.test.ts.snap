// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Environment empty global graphcool rc 1`] = `
Array [
  Cluster {
    "baseUrl": "https://graphcool-eu1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-eu1",
  },
  Cluster {
    "baseUrl": "https://graphcool-us1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-us1",
  },
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterSecret": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment interpolates env vars 1`] = `
Array [
  Cluster {
    "baseUrl": "https://graphcool-eu1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-eu1",
  },
  Cluster {
    "baseUrl": "https://graphcool-us1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-us1",
  },
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterSecret": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment loads multiple cluster definitions correctly + gives cluster by name 1`] = `
Array [
  Cluster {
    "baseUrl": "https://graphcool-eu1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-eu1",
  },
  Cluster {
    "baseUrl": "https://graphcool-us1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-us1",
  },
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterSecret": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment loads multiple cluster definitions correctly + gives cluster by name 2`] = `undefined`;

exports[`Environment non-existent global graphcool rc 1`] = `
Array [
  Cluster {
    "baseUrl": "https://graphcool-eu1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-eu1",
  },
  Cluster {
    "baseUrl": "https://graphcool-us1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-us1",
  },
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterSecret": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`Environment persists .graphcoolrc correctly 1`] = `
"graphcool-1.0:
  clusters:
    cluster:
      host: 'http://localhost:60000'
      clusterSecret: ''
"
`;

exports[`Environment persists .graphcoolrc correctly 2`] = `
Array [
  Cluster {
    "baseUrl": "https://graphcool-eu1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-eu1",
  },
  Cluster {
    "baseUrl": "https://graphcool-us1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-us1",
  },
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterSecret": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
  Cluster {
    "baseUrl": "http://localhost:60000",
    "clusterSecret": "",
    "local": true,
    "name": "cluster",
  },
]
`;

exports[`Environment sets the platform token correctly 1`] = `
Array [
  Cluster {
    "baseUrl": "https://graphcool-eu1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-eu1",
  },
  Cluster {
    "baseUrl": "https://graphcool-us1.graphcool.cloud",
    "clusterSecret": undefined,
    "local": false,
    "name": "graphcool-us1",
  },
  Cluster {
    "baseUrl": "https://database-beta.graph.cool",
    "clusterSecret": undefined,
    "local": false,
    "name": "shared-public-demo",
  },
]
`;

exports[`graphcool definition dont load yml with secret and env var in args 1`] = `[Error: Please either provide a secret in your graphcool.yml or disableAuth: true]`;

exports[`graphcool definition load basic yml, provide cluster 1`] = `
Object {
  "cluster": "local",
  "datamodel": Array [
    "datamodel.graphql",
  ],
  "schema": "schemas/database.graphql",
  "secret": "some-secret",
  "service": "jj",
  "stage": "dev",
}
`;

exports[`graphcool definition load basic yml, provide cluster 2`] = `undefined`;

exports[`graphcool definition load yml with disableAuth: true 1`] = `
Object {
  "cluster": "local",
  "datamodel": Array [
    "datamodel.graphql",
  ],
  "disableAuth": true,
  "schema": "schemas/database.graphql",
  "service": "jj",
  "stage": "dev",
}
`;

exports[`graphcool definition load yml with disableAuth: true 2`] = `null`;

exports[`graphcool definition load yml with injected env var 1`] = `
Object {
  "cluster": "local",
  "datamodel": Array [
    "datamodel.graphql",
  ],
  "schema": "schemas/database.graphql",
  "secret": "some-secret",
  "service": "jj",
  "stage": "dev",
}
`;

exports[`graphcool definition load yml with injected env var 2`] = `
Array [
  "some-secret",
]
`;

exports[`graphcool definition load yml with secret and env var 1`] = `
Object {
  "cluster": "local",
  "datamodel": Array [
    "datamodel.graphql",
  ],
  "schema": "schemas/database.graphql",
  "secret": "this-is-a-long-secret",
  "service": "jj",
  "stage": "dev",
}
`;

exports[`graphcool definition load yml with secret and env var in .env 1`] = `
Object {
  "cluster": "local",
  "datamodel": Array [
    "datamodel.graphql",
  ],
  "schema": "schemas/database.graphql",
  "secret": "this-is-very-secret,and-comma,seperated",
  "service": "jj",
  "stage": "dev",
}
`;

exports[`graphcool definition load yml with secret and env var in .env 2`] = `
Array [
  "this-is-very-secret",
  "and-comma",
  "seperated",
]
`;

exports[`graphcool definition throw when no secret or disable auth provided 1`] = `[Error: Please either provide a secret in your graphcool.yml or disableAuth: true]`;

exports[`graphcool definition throws when stages key apparent 1`] = `
[Error: Invalid graphcool.yml file
graphcool.yml should NOT have a "stages" property anymore. Stages are now just provided as CLI args.
Read more here: https://goo.gl/SUD5i5]
`;
